client/components/header-nav/index.jsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { findDOMNode } from 'react-dom';
+import classNames from 'classnames';
+
+import Gridicon from 'gridicons';
+import Item from 'components/header-nav/item';
+
+export default class HeaderNav extends Component {
+	static propTypes = {
+		options: PropTypes.arrayOf(
+			PropTypes.shape( {
+				label: PropTypes.string.isRequired,
+				icon: PropTypes.string,
+				link: PropTypes.string
+			} )
+		)
+	};
+
+	static defaultProps = {
+		options: []
+	}
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selected: props.href,
```

Comment:
You're right, now that the current URI is passed as a property there is no need for ```state.selected```.